docs/steps.rst

```diff
@@ -28,3 +61,107 @@ Tools for step writing
 .. automodule:: aloe.tools
     :members:
 
+Writing good BDD steps
+----------------------
+
+It's very easy with BDD testing to accidentally reinvent Python testing
+using a pseudo-language. Doing so removes much of the point of using BDD
+testing in the first place, so here is some advice to help write better BDD
+steps.
+
+1.  **Be declarative not imperative**
+
+    If you find yourself specifying implementation details of your application
+    that aren't important to your behaviours, abstract them into a declarative
+    step.
+
+    Imperative:
+
+    .. code-block:: gherkin
+
+        When I fill in username with "danni"
+        And I fill in password with "secret"
+        And I press "Log on"
+        And I wait for AJAX to finish
+
+    Declarative:
+
+    .. code-block:: gherkin
+
+        When I log on as "danni" with password "secret"
+
+    You can use :meth:`Step.behave_as` to write a step that chains up several
+    smaller steps.
+
+2.  **Avoid conjunctions in steps**
+
+    If you're writing a step that contains an `and` or other conjunction
+    consider breaking your step into two.
+
+    You can pass state between steps using :attr:`world`.
+
+3.  **Describe behaviours, not implementation**
+
+    This is especially true when setting up the initial state for a test.
+
+    Implementation:
+
+    .. code-block:: gherkin
+
+        Given the following flags are set:
+            | flags                      |
+            | user_registration_disabled |
+            | user_export_disabled       |
```

Comment:
Also add a suggestion of moving steps that are needed by _all_ tests into a `@before.each_example`.